### PR TITLE
fix: Connect selected calendar to credential

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

Stacked on #11281

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Connect a calendar app
- Select calendars to check for availability
- In the DB the selected calendars should be connected to the credential 

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/calendars/CalendarSwitch.tsx [hunk 1]
```diff
@@ -40,7 +41,7 @@ const CalendarSwitch = (props: ICalendarSwitchProps) => {
           headers: {
             "Content-Type": "application/json",
           },
-          body: JSON.stringify(body),
+          body: JSON.stringify({ ...body, credentialId }),
```
Comment (joeauyeung) [inline packages/features/calendars/CalendarSwitch.tsx:44]: Only the `POST` request needs the credentialId

File: packages/prisma/schema.prisma [hunk 2]
```diff
@@ -448,10 +449,12 @@ model Availability {
...
 model SelectedCalendar {
-  user        User   @relation(fields: [userId], references: [id], onDelete: Cascade)
-  userId      Int
-  integration String
-  externalId  String
+  user         User        @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId       Int
+  integration  String
+  externalId   String
+  credential   Credential? @relation(fields: [credentialId], references: [id], onDelete: Cascade)
```
Comment (joeauyeung) [inline packages/prisma/schema.prisma:456]: When deleting the credential, it should cascade and delete the calendar as well.

File: packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts [hunk 3]
```diff
@@ -372,4 +340,46 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
...
+      where: {
+        userId: user.id,
+        integration: credential.type as string,
+      },
+    });
+
+    if (selectedCalendars.length) {
+      const calendar = await getCalendar(credential);
+
+      const calendars = await calendar?.listCalendars();
```
Comment (joeauyeung) [inline packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts:357]: Backwards compatibility for selected calendars. Eventually, once all calendars are connected to a credential, we won't have to make an API call to a 3rd party to list all calendars.

**Review by keithwillcode — DISMISSED** [review]
Can we add automated tests around this?

**Review by emrysal — DISMISSED** [review]
Lgtm, added issue for follow up @keithwillcode https://github.com/calcom/cal.com/issues/11310

**Review by emrysal — APPROVED** [review]
Re-approve after ts fixes

**Pull request merged**